File: app/src/main/java/com/kanj/apps/topbanner/OthersFixedHeightBanner.java
package com.kanj.apps.topbanner;

import android.content.Context;
import android.view.View;
import com.kanj.apps.topbanner.base.FixedHeightBanner;
import com.kanj.apps.topbanner.core.OthersBannerCallbacks;

/**
 * Created by Kanj Narayan on 22/08/17.
 */

public class OthersFixedHeightBanner extends FixedHeightBanner<OthersBannerCallbacks> {
    private static final int HEIGHT_BANNER_DP = 200;
    private static final int PADDING_BANNER_DP = 15;

    public OthersFixedHeightBanner(Context context, OthersBannerCallbacks callbacks) {
        super(context, callbacks);
    }

    @Override
    protected int getHeightInDp() {
        return HEIGHT_BANNER_DP;
    }

    @Override
    protected void setupView() {
        findViewById(R.id.banner_text).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                bannerCallbacks.bannerTextClicked();
            }
        });
        findViewById(R.id.banner_close).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                bannerCallbacks.onBannerCloseClicked();
            }
        });
    }

    @Override
    protected int getTopPaddingInDp() {
        return PADDING_BANNER_DP;
    }

    @Override
    protected int getBottomPaddingInDp() {
        return PADDING_BANNER_DP;
    }

    @Override
    protected int getLeftPaddingInDp() {
        return PADDING_BANNER_DP;
    }

    @Override
    protected int getRightPaddingInDp() {
        return PADDING_BANNER_DP;
    }

    @Override
    protected int getBannerContentLayout() {
        return R.layout.view_other_banner;
    }

    @Override
    protected int getBannerBackgroundColor() {
        return R.color.colorPrimaryDark;
    }
}

File: app/src/main/java/com/kanj/apps/topbanner/OthersFixedHeightBanner.kt
package com.kanj.apps.topbanner

import android.content.Context
import com.kanj.apps.topbanner.base.FixedHeightBanner
import com.kanj.apps.topbanner.core.OtherBannerScene
import com.kanj.apps.topbanner.core.OthersBannerCallbacks
import kotlinx.android.synthetic.main.view_other_banner.view.*

const val HEIGHT_BANNER_DP = 200
const val PADDING_BANNER_DP = 15

class OthersFixedHeightBanner(context: Context, callbacks: OthersBannerCallbacks)
    : FixedHeightBanner<OtherBannerScene, OthersBannerCallbacks>(context, callbacks), OtherBannerScene {

    override fun getHeightInDp() = HEIGHT_BANNER_DP

    override fun setupView() {
        // lambda arg can be written outside/without ()
        banner_text.setOnClickListener { bannerCallbacks.bannerTextClicked() }
        banner_close.setOnClickListener { bannerCallbacks.onBannerCloseClicked() }
    }

    override fun getTopPaddingInDp() = PADDING_BANNER_DP

    override fun getBottomPaddingInDp() = PADDING_BANNER_DP

    override fun getLeftPaddingInDp() = PADDING_BANNER_DP

    override fun getRightPaddingInDp() = PADDING_BANNER_DP

    override fun getBannerContentLayout() = R.layout.view_other_banner

    override fun getBannerBackgroundColor() = R.color.colorPrimaryDark
}